### ISSUE TITLE
Installation Failure - python version 3.9

### ISSUE THREAD

**Issue by adiegel** [firedrakeproject/firedrake#1974]
I recently updated python to version 3.9. This, of course, broke my firedrake install and, in an attempt to fix it, I simply tried reinstalling once I was directed to the branch located at https://github.com/firedrakeproject/firedrake/pull/1948. Unfortunately, the installation failed during the Petsc installation step. I am attaching my installation log here and hoping someone can help me. 

Thanks!
[firedrake-install.log](https://github.com/firedrakeproject/firedrake/files/5973470/firedrake-install.log)

**Comment by danshapero** [comment]
It looks like this failed in trying to install the Trilinos ML package. Can you attach the PETSc make log? Going off of the paths in your Firedrake installation log, this file should be at /Users/aed388/firedrake/src/petsc/make.log. Alternatively you could navigate to the Firedrake install directory and run `find ./ -name "make.log"`.

At the same time, you could try installing again but pass the option `--minimal-petsc` to the Firedrake install script. This should skip over trying to install ML.

**Comment by adiegel** [comment]
Thanks for getting back to me so quickly. I tried to run it with the --minimal-petsc and it failed again. I'm attaching the new install log along with the petsc config log.

The exact call I used: python3 firedrake-install --package-branch firedrake JDBetteridge/python3.9 --minimal-petsc

[firedrake-install.log](https://github.com/firedrakeproject/firedrake/files/5973808/firedrake-install.log)
[configure.log](https://github.com/firedrakeproject/firedrake/files/5973813/configure.log)

**Comment by danshapero** [comment]
It looks like the failure is on line 34074 of configure.log:

    clang: error: invalid version number in '-mmacosx-version-min=11.2'

I googled this and found [this SO thread](https://stackoverflow.com/questions/63972113/big-sur-clang-invalid-version-error-due-to-macosx-deployment-target). One suggestion there was to do the following:

    export SYSTEM_VERSION_COMPAT=1

So that might be worth trying as it doesn't involve installing or uninstalling anything. Failing that, we could try some of the other advice in that thread.

**Comment by adiegel** [comment]
Okay. I finally was able to install successfully. The upgrade to Big Sur did indeed break something with clang. I was able to fix this by following the instructions from the thread posted above by danshapero. Thanks! 

Then, I unlinked python 3.9 and forced python 3.8 to be used instead by following:

brew unlink python@3.9
brew link --force --overwrite python@3.8

I also used the following commands (put together from several other branches on the help boards here):

export PETSC_CONFIGURE_OPTIONS="--COPTFLAGS='-g -O3 -Wno-implicit-function-declaration'"
export PATH=/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/TeX/texbin
/usr/local/bin/python3 firedrake-install --minimal-petsc

Not sure if all of that was required, but it's the combination that worked! Thanks for your help!
